src/runtime_src/hip/api/hip_memory.cpp

```diff
@@ -75,7 +76,7 @@ namespace xrt::core::hip
     assert(device_ptr);
 
     auto hip_mem = memory_database::instance().get_hip_mem_from_addr(host_ptr).first;
-    throw_invalid_value_if(!hip_mem, "Error getting device pointer from host_malloced memory!");
+    throw_invalid_value_if(!hip_mem, "Error getting device pointer from host pointer " + xrt_core::utils::to_hex(host_ptr) + " in " + __func__);
```

Comment:
Too expensive to call `to_hex()` and string append on success path.  
I don't know the signature of `throw_invalid_value_if`, but it should take no more than condition and `const char*`.

Reply:
The ternary case you show in principle looks cheap.   

Calling throw_invalid_value_if(bool, const std::string&) is not kosher on success path.  You will incur at least one string ctor for the empty string; cheap, albeit annoying, especially when debugging.   I vaguer that the signature should be throw_invalid_value_if(bool, const char* msg).  If you need any fancy error message, I think it ought to be a special case.

Reply:
type of err_msg has been changed to const char*.